### PR TITLE
Refactor evaluation data into object.

### PR DESCRIPTION
**Pull request by pmtischler**
This is in prep for adding more evaluation data that will be exported
(e.g. total mission score), and for a more complex export which will
include json (post-analysis), csv (downstream scoring), and tex (team
feedback).

For #218

### CODE REVIEW

File: server/auvsi_suas/models/mission_config.py [hunk 1]
```diff
@@ -232,89 +212,74 @@ def evaluate_teams(self, users=None):
             logger.info('Evaluation starting for user: %s.' % user.username)
 
             # Start the evaluation data structure.
-            eval_data = results.setdefault(user, {})
-            warnings = []
-            eval_data['warnings'] = warnings
+            results[user] = MissionEval()
```
Comment (yogishadixit) [inline server/auvsi_suas/models/mission_config.py]: Nit: switch these order of these statements:
```
user_eval = MissionEval()
user_eval.team = user
results[user] = user_eval
```

File: server/auvsi_suas/models/mission_config_test.py [hunk 2]
```diff
@@ -204,6 +205,67 @@ def test_satisfied_waypoints(self):
...
+                'obst_collision_moving': {2: True},
+                'targets': {
+                    'auto': 'test',
+                    'manual': 'test',
+                },
+            }, self.eval.json())
+
+    def test_csv(self):
+        self.assertEqual(
+            {
```
Comment (yogishadixit) [inline server/auvsi_suas/models/mission_config_test.py]: I'm confused. How is this a CSV?

File: server/auvsi_suas/models/mission_config.py [hunk 3]
```diff
@@ -486,3 +451,79 @@ def kml(self, kml):
...
             name='Stationary Obstacles')
+
+
+class MissionEval(object):
+    """Mission evaluation data for a single team.
+
+    Attributes:
+        team: The team for which this is an evaluation.
+        warnings: List of warnings generated during evaluation.
+        uas_telemetry_time_max: Slowest time of telemetry sent.
```
Comment (yogishadixit) [inline server/auvsi_suas/models/mission_config.py]: s/time/rate

File: server/auvsi_suas/models/mission_config.py (hunk 1 above)
Comment (pmtischler) [inline server/auvsi_suas/models/mission_config.py]: Done.

File: server/auvsi_suas/models/mission_config.py [hunk 4]
```diff
@@ -232,89 +212,74 @@ def evaluate_teams(self, users=None):
...
                 'auto': [t for t in user_targets if t.autonomous],
             }
             for target_set, targets in target_sets.iteritems():
                 evaluator = TargetEvaluator(targets, self.targets.all())
-                eval_data['targets'][target_set] = evaluator.evaluation_dict()
+                user_eval.targets[target_set] = evaluator.evaluation_dict()
 
             # Determine interop telemetry rates.
-            uas_telemetry_times = UasTelemetry.rates(
+            uas_telemetry_time_max, uas_telemetry_time_avg = UasTelemetry.rates(
```
Comment (pmtischler) [inline server/auvsi_suas/models/mission_config.py]: That doesn't fit in the width.

File: server/auvsi_suas/models/mission_config.py [hunk 5]
```diff
@@ -486,3 +451,79 @@ def kml(self, kml):
         # Stationary Obstacles
         stationary_obstacles_folder = kml_folder.newfolder(
             name='Stationary Obstacles')
+
+
+class MissionEval(object):
+    """Mission evaluation data for a single team.
+
+    Attributes:
+        team: The team for which this is an evaluation.
```
Comment (pmtischler) [inline server/auvsi_suas/models/mission_config.py]: Done.

File: server/auvsi_suas/models/mission_config_test.py (hunk 2 above)
Comment (pmtischler) [inline server/auvsi_suas/models/mission_config_test.py]: The same as the above isn't technically json :) It's the format before explicitly writing to the end format.

Json requires only primitives and list/dict containers.
Csv requires only dict with primitives (keys are column, value is row value).

File: server/auvsi_suas/views/auvsi_admin/evaluate_teams.py [hunk 6]
```diff
@@ -63,40 +62,25 @@ class EvaluateTeamsJson(EvaluateTeamsBase):
...
         col_headers = set()
         for col_data in user_col_data.values():
             for header in col_data.keys():
                 col_headers.add(header)
         col_headers = sorted(col_headers)
 
-        # Write output
+        # Write output.
         csv_output = cStringIO.StringIO()
         writer = csv.DictWriter(csv_output, fieldnames=col_headers)
```
Comment (pmtischler) [inline server/auvsi_suas/views/auvsi_admin/evaluate_teams.py]: @yogishadixit This is why it takes a `dict`.

**Pull request merged**